### PR TITLE
Fixed undefined errors

### PR DESCRIPTION
**Pull request by mariusvw**

### CODE REVIEW

File: lib/exe/js.php [hunk 1]
```diff
@@ -173,7 +173,7 @@ function js_load($file){
         // is it a include_once?
         if($match[1]){
             $base = utf8_basename($ifile);
-            if($loaded[$base]){
+            if(array_key_exists($base, $loaded) && $loaded[$base] === true){
```
Comment (Chris--S) [inline lib/exe/js.php]: Do you have a problem with the simpler (&abetted) !empty()?

File: lib/exe/js.php [hunk 2]
```diff
@@ -419,9 +419,9 @@ function js_compress($s){
 
             // Only consider deleting whitespace if the signs before and after
             // are not equal and are not an operator which may not follow itself.
-            if ((!$lch || $s[$i+1] == ' ')
+            if (isset($s[$i+1]) && ((!$lch || $s[$i+1] == ' ')
```
Comment (Chris--S) [inline lib/exe/js.php]: This isn't functionally equivalent.  plus, is the more sensible test, $i+1 < $slen ?

File: lib/exe/js.php (hunk 1 above)
Comment (mariusvw) [inline lib/exe/js.php]: Empty would do the trick also, I just prefer strict checking, if thats over done, I'll replace it with a !empty... Let me know what you guys prefer :)

File: lib/exe/js.php (hunk 2 above)
Comment (mariusvw) [inline lib/exe/js.php]: You are right, this works better.

**Pull request merged**